### ISSUE TITLE
Are there any examples that load a mesh from a obj file?

### ISSUE THREAD

**Issue by ForceConstant** [owl-project/owl#162]
I am a little confused on the best way to do this, so would like to see an example.

Thanks.

**Comment by ingowald** [comment]
There is an example that reads in an OBJ file (samples/advanced/optix7course). However, OWL conceptually is not - nor is it intended to be - a "scene graph" API with different model format importers, etc; think of it as an API that, assuming you already have some data in memory - lets you more easily get that model into optix.

**Comment by ingowald** [comment]
In general for OBJ files, Syoyo Fujita's tinyobj (https://github.com/tinyobjloader/tinyobjloader/blob/master/tiny_obj_loader.h) is probably the easiest way to load OBJ files; it's also what this sample is using.

**Comment by ForceConstant** [comment]
@ingowald Thanks for the pointers. So I am looking at optix7course example, and have it loading my obj but have some questions on how some stuff works, so I can see about adding my own code. 

So I need to calculate and save a depth buffer that is calculated using optixGetRayTmax().  So in host code I added 
```
    depths = owlDeviceBufferCreate(context,OWL_FLOAT,fbSize.x*fbSize.y,nullptr);
    owlParamsSetBuffer(launchParams,"depths",depths);
```

And then in the render function, can I just access this saved buffer using 'auto fb = owlBufferGetPointer(depths , 0);'

**Comment by bipul-mohanto** [comment]
It is actually the question from my side, `how some stuff works!` I have slightly modified `optix7course` functions what I need. It is working in some way (window is created), but no models are there. I have basic understanding how camera works, but still understanding and modifying some functions and data a bit difficult for me.

**Comment by ingowald** [comment]
@ForceConstant Almost, but not entirely. "owlBufferGetPointer" is  *host* function, so you could call it on the host to get the device address from a buffer. On the device side, you want to add a new pointer to that depth buffer to the "LaunchParams" struct (see LaunchParams.h in that same subdir) - say, you call it "float *depthBuffer" - then "export" that pointer when you declare the launch params in the same way that the color buffer ("LaunchParams::fbColor") is being handled.

Basically, just follow how "fbColor" is being used and treated on both host and device code.

**Comment by ingowald** [comment]
You may also want to read the wiki (https://github.com/owl-project/owl/wiki) in particular the page on launch params (https://github.com/owl-project/owl/wiki/Launch-Parameters)
